deriva/core/ermrest_model.py

```diff
@@ -93,39 +138,31 @@ def create_table(self, catalog, table_def):
         tname = table_def['table_name']
         if tname in self.tables:
             raise ValueError('Table %s already exists.' % tname)
-        r = catalog.post(
+        r = self.catalog.post(
             '%s/table' % self.uri_path,
             json=table_def,
         )
         r.raise_for_status()
-        newtable = Table(self.name, tname, r.json())
+        newtable = Table(self, tname, r.json())
         self.tables[tname] = newtable
+        self.model.digest_fkeys()
         return newtable
 
-    def delete(self, catalog, model=None):
+    def delete(self):
```

Comment:
how about `drop` instead of `delete` for the model change operations that way `delete` can be used unambiguously in datapath for tuple deletes?